pkg/kubelet/cm/dra/manager.go

```diff
@@ -66,112 +66,108 @@ func NewManagerImpl(kubeClient clientset.Interface, stateFileDirectory string) (
-		for range container.Resources.Claims {
-			for i, podResourceClaim := range pod.Spec.ResourceClaims {
-				claimName := resourceclaim.Name(pod, &pod.Spec.ResourceClaims[i])
-				klog.V(3).InfoS("Processing resource", "claim", claimName, "pod", pod.Name)
-
-				// Resource is already prepared, add pod UID to it
-				if claimInfo := m.cache.get(claimName, pod.Namespace); claimInfo != nil {
-					// We delay checkpointing of this change until this call
-					// returns successfully. It is OK to do this because we
-					// will only return successfully from this call if the
-					// checkpoint has succeeded. That means if the kubelet is
-					// ever restarted before this checkpoint succeeds, the pod
-					// whose resources are being prepared would never have
-					// started, so it's OK (actually correct) to not include it
-					// in the cache.
-					claimInfo.addPodReference(pod.UID)
-					continue
-				}
+	for i, podResourceClaim := range pod.Spec.ResourceClaims {
+		claimName := resourceclaim.Name(pod, &pod.Spec.ResourceClaims[i])
+		klog.V(3).InfoS("Processing resource", "claim", claimName, "pod", pod.Name)
+
+		// Resource is already prepared, add pod UID to it
+		if claimInfo := m.cache.get(claimName, pod.Namespace); claimInfo != nil {
+			// We delay checkpointing of this change until this call
+			// returns successfully. It is OK to do this because we
+			// will only return successfully from this call if the
+			// checkpoint has succeeded. That means if the kubelet is
+			// ever restarted before this checkpoint succeeds, the pod
+			// whose resources are being prepared would never have
+			// started, so it's OK (actually correct) to not include it
+			// in the cache.
+			claimInfo.addPodReference(pod.UID)
+			continue
+		}
 
-				// Query claim object from the API server
-				resourceClaim, err := m.kubeClient.ResourceV1alpha2().ResourceClaims(pod.Namespace).Get(
-					context.TODO(),
-					claimName,
-					metav1.GetOptions{})
-				if err != nil {
-					return fmt.Errorf("failed to fetch ResourceClaim %s referenced by pod %s: %+v", claimName, pod.Name, err)
-				}
+		// Query claim object from the API server
+		resourceClaim, err := m.kubeClient.ResourceV1alpha2().ResourceClaims(pod.Namespace).Get(
+			context.TODO(),
+			claimName,
+			metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to fetch ResourceClaim %s referenced by pod %s: %+v", claimName, pod.Name, err)
+		}
 
-				// Check if pod is in the ReservedFor for the claim
-				if !resourceclaim.IsReservedForPod(pod, resourceClaim) {
-					return fmt.Errorf("pod %s(%s) is not allowed to use resource claim %s(%s)",
-						pod.Name, pod.UID, podResourceClaim.Name, resourceClaim.UID)
-				}
+		// Check if pod is in the ReservedFor for the claim
+		if !resourceclaim.IsReservedForPod(pod, resourceClaim) {
+			return fmt.Errorf("pod %s(%s) is not allowed to use resource claim %s(%s)",
+				pod.Name, pod.UID, podResourceClaim.Name, resourceClaim.UID)
+		}
```

Comment:
To avoid a future rebase issue, you can remove `podResourceClaim` from the loop variables and change the reference to `podResourceClaim` to `claimName`.

Reply:
thanks. will do unless this PR is merged before it requires a rebase.

Reply:
btw, last time I didn't rebase. I took the code from master and removed 2 loops from it as rebasing directly created conflict and I thought it's more clear to start from scratch then to resolve it.